server/log_rpc_server_test.go

```diff
@@ -1132,192 +1132,167 @@ func TestGetSequencedLeafCount(t *testing.T) {
-	// getConsistencyProofRequest44 - tests an edge condition we used to reject but now support
-	// for compatibility with the older C++ log servers.
-	testCases := []trillian.GetConsistencyProofRequest{getConsistencyProofRequest7, getConsistencyProofRequest44}
-	nodeIDs := [][]storage.NodeID{nodeIdsConsistencySize4ToSize7, {}}
-	hashes := [][][]byte{{[]byte("nodehash")}, {}}
-	nodes := [][]storage.Node{{{NodeID: stestonly.MustCreateNodeIDForTreeCoords(2, 1, 64), NodeRevision: 3, Hash: []byte("nodehash")}}, {}}
-
-	for i := range testCases {
+	for _, test := range tests {
 		mockStorage := storage.NewMockLogStorage(ctrl)
 		mockTx := storage.NewMockLogTreeTX(ctrl)
-		mockStorage.EXPECT().SnapshotForTree(gomock.Any(), testCases[i].LogId).Return(mockTx, nil)
-
-		mockTx.EXPECT().LatestSignedLogRoot(gomock.Any()).Return(signedRoot1, nil)
-		mockTx.EXPECT().ReadRevision().Return(signedRoot1.TreeRevision)
-		mockTx.EXPECT().GetMerkleNodes(gomock.Any(), revision1, nodeIDs[i]).Return(nodes[i], nil)
-		mockTx.EXPECT().Commit().Return(nil)
-		mockTx.EXPECT().Close().Return(nil)
+		if !test.noSnap {
+			mockStorage.EXPECT().SnapshotForTree(gomock.Any(), test.req.LogId).Return(mockTx, test.snapErr)
+		}
+		if !test.noRoot {
+			mockTx.EXPECT().LatestSignedLogRoot(gomock.Any()).Return(signedRoot1, test.rootErr)
+		}
+		if !test.noRev {
+			mockTx.EXPECT().ReadRevision().Return(signedRoot1.TreeRevision)
+		}
+		if test.nodeIDs != nil {
+			mockTx.EXPECT().GetMerkleNodes(gomock.Any(), revision1, test.nodeIDs).Return(test.nodes, test.getNodesErr)
+		}
+		if !test.noCommit {
+			mockTx.EXPECT().Commit().Return(test.commitErr)
+		}
+		if !test.noClose {
+			mockTx.EXPECT().Close().Return(nil)
+		}
 
 		registry := extension.Registry{
-			AdminStorage: mockAdminStorage(ctrl, testCases[i].LogId),
+			AdminStorage: mockAdminStorage(ctrl, test.req.LogId),
 			LogStorage:   mockStorage,
 		}
 		server := NewTrillianLogRPCServer(registry, fakeTimeSource)
-
-		response, err := server.GetConsistencyProof(context.Background(), &testCases[i])
-		if err != nil {
-			t.Fatalf("failed to get consistency proof: %v", err)
-		}
-
-		// Ensure we got the expected proof
-		expectedProof := trillian.Proof{
-			LeafIndex: 0,
-			Hashes:    hashes[i],
-		}
-		if !proto.Equal(response.Proof, &expectedProof) {
-			t.Fatalf("expected proof: %v but got: %v", expectedProof, response.Proof)
+		response, err := server.GetConsistencyProof(context.Background(), &test.req)
+
+		if test.wantErr {
+			if err == nil {
+				t.Errorf("TestGetConsistencyProof(%+v)=nil, want: err", test.req)
+			}
+		} else {
+			if err != nil {
+				t.Errorf("TestGetConsistencyProof(%+v)=%v, want: nil", test.req, err)
+			}
+			// Ensure we got the expected proof.
+			wantProof := trillian.Proof{
+				LeafIndex: 0,
+				Hashes:    test.wantHashes,
+			}
+			if got, want := response.Proof, &wantProof; !proto.Equal(got, want) {
+				t.Errorf("TestGetConsistencyProof got: %v, want: %v", got, want)
```

Comment:
be nice to be in the same format as the errors above

Reply:
Done.